internal/pkg/earpointer/analysis.go

```diff
@@ -28,9 +28,20 @@ import (
 
 	"golang.org/x/tools/go/analysis"
 	"golang.org/x/tools/go/analysis/passes/buildssa"
+	"golang.org/x/tools/go/callgraph"
+	"golang.org/x/tools/go/callgraph/static"
 	"golang.org/x/tools/go/ssa"
 )
 
+var (
+	contextK = 0 // the number of call sites in each context
+)
+
+func init() {
+	Analyzer.Flags.IntVar(&contextK, "contextK", 0,
+		`the K value (default=0) in context sensitivity. Currently only K=0 or K=1 are supported`)
+}
+
 // Analyzer traverses the packages and constructs an EAR partitions
 // unifying all the IR elements in these packages.
 var Analyzer = &analysis.Analyzer{
```

Comment:
This is a more general point about the previous CLs that adding the flag highlights. I have not heard a good reason why Analyzer should be in the earpointer package (other than organization being slightly harder). This mixes concepts. I have not seen an argument that it needs to access internals. I suspect you will need to undo this going forward so that this can be reused outside of the context of an analysis.Analyzer (i..e. next week.)

Reply:
Now the main analysis part has been separated from the pass part, and the analysis can be used by other passes. In Levee, we don't need to put them into separate package though since its only usage is through the Pass interface.